### PR TITLE
Add tests for global categories, releases, fix bugged tests

### PR DESCRIPTION
**Pull request by kevinsekuj**
**Additions**
- Added further unit tests for new releases, passing limit parameter with minimum and maximum values of 1 and 50
- Added further unit tests for categories, omitting country code to test global releases

**Fixes**
- Fixed potential error where `found` variable in `test_artist_related_artists` _tests/integration/non_user_endpoints/test.py_ is undefined if for loop never evaluates to true
- Fixed false positive test `test_new_releases` which looks up the wrong property of the JSON response object and always evaluates to true

### CODE REVIEW

**Comment by kevinsekuj** [comment]
@stephanebruckert any ideas why the job for python build fails?

**Comment by stephanebruckert** [comment]
@kevinsekuj I just fixed it on the default branch

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,18 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 -   Incorrect category_id input for test_category
 -   Assertion value for test_categories_limit_low and test_categories_limit_high
 
+## [2.21.1] - 2022-11-26
```
Comment (stephanebruckert) [inline CHANGELOG.md:19]: We can leave the changes in `Unreleased` rather than create a new version just yet

Comment (kevinsekuj) [reply CHANGELOG.md]: definitely, thanks👍🏻

File: tests/integration/user_endpoints/test.py [hunk 2]
```diff
@@ -364,10 +364,18 @@ def test_categories_country(self):
...
+        self.assertGreater(len(response['categories']), 0)
+
     def test_categories_locale(self):
         response = self.spotify.categories(locale='en_US')
         self.assertGreater(len(response['categories']), 0)
 
+    def test_categories_locale_global(self):
+        response = self.spotify.categories()
+        self.assertGreater(len(response['categories']), 0)
+
```
Comment (stephanebruckert) [inline tests/integration/user_endpoints/test.py:378]: this is the same test as `test_categories_global`?

**Review by stephanebruckert — APPROVED** [review]
Thanks!

**Pull request merged**